featuretools/entityset/entityset.py

```diff
@@ -485,6 +485,10 @@ def entity_from_dataframe(self,
         if time_index is not None and time_index == index:
             raise ValueError("time_index and index cannot be the same value, %s" % (time_index))
 
+        if vtypes.DatetimeTimeIndex in variable_types.values() and time_index is None:
+            var_name = variable_types.keys()[variable_types.values().index(vtypes.DatetimeTimeIndex)]
```

Comment:
This works but I think we can clean it up a bit

We can loop through `variable_types.items()` once and raise an error the first time we encounter the problem.